website/docs/development/release.md

```diff
@@ -6,9 +6,43 @@ sidebar_label: Release process
 
 The release process may be automated in the future.
 
-- Checkout master
-- Update the Hydra version in `hydra/__init__.py`
-- Update NEWS.md with towncrier
-- Create a pip package for hydra-core: `python setup.py sdist bdist_wheel`
+The versioning strategy is that the repo is configured to be on the next version and the version should be bumped AFTER the release, not before.
+
+One should check that the version is the correct one against pypi before releasing.
+
+### Hydra Release Process
+
+- Change branch to release branch and git pull to make sure on top of latest commit.
+- Checkout a local branch for releasing.
+- Check the version in `hydra/__init__.py`. This should be one version higher than the version available on [PyPI](https://pypi.org/project/hydra-core/). If it is the same version as PyPI, bump the version locally.
+- Create draft release notes using `towncrier build --draft`.
+- Make sure the draft notes look correct.
+- Update NEWS.md using `towncrier build`.
```

Comment:
I don't know what version of towncrier you are using, but 19.2.0 (latest) does not have a build command.